FFM-7066: DO NOT MERGE (awaiting GA) All environments page (#1226)
harness/developer-hub: docs/feature-flags/ff-using-flags/filtering-flags-by-state.md

---

> title: View and Filter Flags

Reviewer: Use sentence case for titles.

Author: Oh wow, I've been following the title style in the docs, but it turns out ALL of the FF titles are wrong >camel case. Let me fix those in one go after I merge this: [FFM-7548](https://harness.atlassian.net/browse/FFM-7548)

description: To help manage your Feature Flags, you can use the filter tiles on the Harness Platform to filter your Flags based on the following states --  Figure 1 --  The Flag overview dashboard State Description All…
sidebar_position: 25
helpdocs_topic_id: 2s5qt02s74
helpdocs_category_id: t5s8pv9gjb
helpdocs_is_private: false
helpdocs_is_published: true
---

The Feature Flags dashboard lets you filter flags by state (for example, **Enabled**) and see the state of flags in different environments.

## Filter flags by state

**To view filter your flags:**

1. In Harness, go to your project, and then select **Feature Flags**.

    The dashboard displays the total number of flags, and the number of flags in each state. For details, go to [Flag states](#flag-states) below. 

    ![A screenshot of the Permanent Flags tile selected to filter for permanent flags.](../static/9-filtering-flags-by-state-01.png)

> 3. Select **All Flags**, or use one of the available filters, for example, **Permanent Flags**.

Reviewer: This should be step 2.

    The dashboard displays the flags for your selected filter.

### Flag states

| **State** | **Description** |
| --- | --- |
| **Enabled Flags** | Flags that are currently toggled on. |
| **Permanent Flags** | Flags you intend to stay in your systems indefinitely and that you marked as permanent when creating them. Permanent flags are never marked as stale. |
| **Recently Changed Flags** | Flags that have been changed in the last 24 hours. Changes include enabling or disabling a flag, or adding new Rules or Targets. |

> | **Active Flags** | Flags that have been evaluated in the last 7 days |

Reviewer: ```suggestion
| **Active Flags** | Flags that have been evaluated in the last 7 days. |
```

> | **Potentially Stale Flags** | Flags are marked as potentially stale if in the past 60 days they haven't been changed or evaluated,their default rules or target rules haven’t been added to or updated, they haven’t been toggled on or off.|

Reviewer: ```suggestion
| **Potentially Stale Flags** | Flags are marked as potentially stale if in the past 60 days they haven't been changed or evaluated, their default rules or target rules haven’t been added to or updated, or they haven’t been toggled on or off.|
```

> | **Potentially Stale Flags** | Flags are marked as potentially stale if in the past 60 days they haven't been changed or evaluated,their default rules or target rules haven’t been added to or updated, they haven’t been toggled on or off.|

Author: Yup, they're capitalized.

:::info note

>  A flag can be marked as both **Active** and **Potentially Stale** if it has been evaluated via an SDK but no other changes have been made in over 60 days. For example, if you evaluated `Flag_A` yesterday using an SDK, but haven’t made any changes on the Harness Platform in over three months, the flag will be marked as Active and Potentially Stale on the Platform.

Reviewer: You used bold on Active and Potentially Stale in the first sentence, but not in the second sentence. 

Also, the last part of the second sentence, "on the Platform". Use "Harness Platform".

:::

## View flag states in different environments

The Feature Flags dashboard can display the flag states and details for a selected environment, or a summary of flag states across all environments.

To view flag states in different environments:

1. In Harness, go to your project, and then select **Feature Flags**.
 

> 1. In **Environments**, choose an environment from the drop-down, or **All Environments**.

Reviewer: ```suggestion
2. In **Environments**, select a specific environment, or **All Environments**.
```
